src/hooks/useWidgetWidth.tsx

```diff
@@ -0,0 +1,20 @@
+import { createContext, PropsWithChildren, useContext } from 'react'
+
+const WidgetWidthContext = createContext<number>(0)
+
+interface WidgetWidthProviderProps {
+  width: number
+}
+
+export function WidgetWidthProvider(props: PropsWithChildren<WidgetWidthProviderProps>) {
```

Comment:
nit: I prefer to destructure: `{ width, children }: PropsWithChildren<WidgetWidthProviderProps> {`